diff --git a/src/frontend/CMakeLists.txt b/src/frontend/CMakeLists.txt
index 70ea393..5a4d049 100644
--- a/src/frontend/CMakeLists.txt
+++ b/src/frontend/CMakeLists.txt
@@ -5,14 +5,7 @@ target_sources(
   frontend.h
 
   impl/memory_trace/loadstore_trace.cpp
-  impl/memory_trace/readwrite_trace.cpp
-
-  impl/processor/simpleO3/simpleO3.cpp
-  impl/processor/simpleO3/core.h      impl/processor/simpleO3/core.cpp
-  impl/processor/simpleO3/llc.h       impl/processor/simpleO3/llc.cpp
-  impl/processor/simpleO3/trace.h     impl/processor/simpleO3/trace.cpp
-
-  impl/external_wrapper/gem5_frontend.cpp
+  impl/memory_trace/pim_loadstore_trace.cpp
 )
 
 target_link_libraries(
