diff --git a/src/base/request.h b/src/base/request.h
index 2c9d406..fa657f8 100644
--- a/src/base/request.h
+++ b/src/base/request.h
@@ -17,8 +17,24 @@ struct Request {
   // 0 = Read, 1 = Write. The device spec defines all others
   struct Type {
     enum : int {
+      // DRAM requests
       Read = 0, 
-      Write,
+      Write = 1,
+
+      // PIM requests
+      PIM_MAC_AB = 4,
+      PIM_MAC_SB = 5,
+      PIM_MAC_PB = 6,
+      PIM_WR_GB = 7,
+      PIM_MV_SB = 8,
+      PIM_MV_GB = 9,
+      PIM_SFM = 10,
+      PIM_SET_MODEL = 11,
+      PIM_SET_HEAD = 12,
+      PIM_ACC = 13,
+      PIM_AF = 14,
+      PIM_EWMUL = 15,
+      PIM_BARRIER = 16
     };
   };
 
