diff --git a/src/dram/CMakeLists.txt b/src/dram/CMakeLists.txt
index 3b781b6..047b3e1 100644
--- a/src/dram/CMakeLists.txt
+++ b/src/dram/CMakeLists.txt
@@ -6,14 +6,15 @@ target_sources(
   
   lambdas/preq.h  lambdas/rowhit.h  lambdas/rowopen.h 
 
-  impl/DDR3.cpp
-  impl/DDR4.cpp
-  impl/DDR4-VRR.cpp
-  impl/DDR5.cpp
-  impl/LPDDR5.cpp
-  impl/HBM.cpp
-  impl/HBM2.cpp
-  impl/HBM3.cpp
+  # impl/DDR3.cpp
+  # impl/DDR4.cpp
+  # impl/DDR4-VRR.cpp
+  # impl/DDR5.cpp
+  # impl/LPDDR5.cpp
+  # impl/HBM.cpp
+  # impl/HBM2.cpp
+  # impl/HBM3.cpp
+  impl/DDR4_PIM.cpp
+  impl/LPDDR5_PIM.cpp
 )
 
 target_link_libraries(
